src/cache/abstract-assignment-cache.ts

```diff
@@ -2,28 +2,29 @@ import { getMD5Hash } from '../obfuscation';
 
 import { LRUCache } from './lru-cache';
 
-export type AssignmentCacheValue = {
-  allocationKey: string;
-  variationKey: string;
-};
-
+/**
+ * Assignment cache keys are only on the subject and flag level, while the entire value is used
+ * for uniqueness checking. This way that if an assigned variation or bandit action changes for a
+ * flag, it evicts the old one. Then, if an older assignment is later reassigned, it will be treated
+ * as new.
+ */
 export type AssignmentCacheKey = {
   subjectKey: string;
   flagKey: string;
 };
 
+export type AssignmentCacheValue = Record<string, string>;
```

Comment:
@rasendubi & @felipecsl cache is now generic

Reply:
We will lose type-safety by using generic `Record`. Would using generics be possible?

```
export type CacheKeyPair<T extends string, U extends string> = {
  [K in T]: string;
} & {
  [K in U]: string;
};

export type AssignmentCacheValue = CacheKeyPair<'allocationKey', 'variationKey'>;
export type BanditCacheValue = CacheKeyPair<'banditKey', 'actionKey'>;

// Usage example
const assignmentCache: AssignmentCacheValue = {
  allocationKey: 'someAllocationKey',
  variationKey: 'someVariationKey'
};

const banditCache: BanditCacheValue = {
  banditKey: 'someBanditKey',
  actionKey: 'someActionKey'
};
```

Reply:
Oh this is an interesting thought! I think this would be possible, but I don't know if we _need_ to be not generic. If they send in something other than bandit key and action key, we'd still hash it and use that to define uniqueness for the entry, which I think is ok. Keeps it simpler and makes it more extensible if we need to cache other stuff going forward.

Reply:
Paired on this and got 80% the way there; will merge then possibly go for the other 20% later.